Change in authorization for create new template for MealPlanner app u…

invisible-pvi opened the pull request:
**Describe the technical changes contained in this PR**
Created a condition to only allow the users who do not have the role of app_user to view the Create New Template.


**Previous behaviour**
Previously, any user of the app would be able to create a new meal plan template, a functionality to be used only by app admins or meal designers.

**New behaviour**
Now, the app allows users with only admin and meal designer roles to create a new meal template.

![EAEB398C-77B7-4021-B8A4-A8AD36B44A44](https://github.com/CivicTechFredericton/mealplanner/assets/156869108/830b59f0-0a04-44ba-a198-8f394474ae38)




**Related issues addressed by this PR**
Fixes #687 

**Have the following been addressed?**
- [ ] Have test cases been created for all of the changes?
- [ ] Do all of the test cases pass?
- [ ] Has the testing been done using the default docker-compose environment?
- [ ] Are documentation changes required?
- [ ] Does this change break or alter existing behaviour?

invisible-wind-pavitra committed:
Change in authorization for create new template for MealPlanner app users

shanthisa commented:
The radio button Search by Template should also not appear for app_user because there are no templates for client users.

invisible-wind-pavitra committed:
Change in authorization for search template for MealPlanner app users

invisible-pvi commented:
Hi Shanthi, I have fixed the second issue based on the above comment and updated the pull request above. Now, it looks like this: 

![EAEB398C-77B7-4021-B8A4-A8AD36B44A44](https://github.com/CivicTechFredericton/mealplanner/assets/156869108/51d8eeb5-2f2a-479f-a60e-f3411fb52c66)

shanthisa reviewed (approved):
LGTM 👍
